lib/address_geocoder/MapApiManager

```diff
@@ -0,0 +1,20 @@
+# Map API Manager don't Know HOW to do every action
+# know WHAT to do and WHO to do
+# know address_geocoder requirement
+
+require 'map_apis/*'
+
+class AddressGeocoder
+  class MapApiManager
+    def initialize(opt = {})
+      # add attributes
+      # also validation inputs
+    end
+
+    def process
+      # call MapApiWrappers
+      # call ReportWriter
```

Comment:
Right before here we are going to have to parse because I doubt that we can compare results without parsing